src/com/webonise/carbuilder/ColorFactory.java

```diff
@@ -0,0 +1,40 @@
+package com.webonise.carbuilder;
+
+public class ColorFactory extends AbstractCarFactory{
+
+	@Override
+	public ITypeOfCar getTypeOfCar(String TypeOfCar) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public ICompany getCompany(String Company) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public IEngine getEngine(String Engine) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public IColor getColor(String color) {
+		if(color == null){
```

Comment:
Please check for empty and string containing only white spaces. Use `isBlank`
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isBlank(java.lang.CharSequence)